meteor/client/ui/Prompter/controller/joycon-device.ts

```diff
@@ -221,28 +228,33 @@ export class JoyConController extends ControllerAbstract {
 		let lastSeenSpeed = 0
 
 		for (const joycon of joycons) {
-			// handle buttons at the same time as evaluating stick input
-			this.handleButtons(joycon)
-
-			// hadle speed input
-			if (joycon.mode === 'L' || joycon.mode === 'R') {
-				// L or R mode
-				if (Math.abs(joycon.axes[0]) > this.deadBand) {
-					if (joycon.mode === 'L') {
-						lastSeenSpeed = joycon.axes[0] * -1 // in this mode, L is "negative"
-					} else if (joycon.mode === 'R') {
-						lastSeenSpeed = joycon.axes[0] * 1.4 // in this mode, R is "positive"
+			// sort/filter by gamepad timestamp to use the most up-to-date input, in order to prevent the "stuck" dead joycon when going from pairs to singles
+			if (joycon.timestamp >= this.timestampOfLastUsedJoyconInput) {
+				// handle buttons at the same time as evaluating stick input
+				this.handleButtons(joycon)
+
+				// hadle speed input
+				if (joycon.mode === 'L' || joycon.mode === 'R') {
+					// L or R mode
+					if (Math.abs(joycon.axes[0]) > this.deadBand) {
+						if (joycon.mode === 'L') {
+							lastSeenSpeed = joycon.axes[0] * -1 // in this mode, L is "negative"
+						} else if (joycon.mode === 'R') {
+							lastSeenSpeed = joycon.axes[0] * 1.4 // in this mode, R is "positive"
+							// factor increased by 1.4 to account for the R joystick being less sensitive than L
+						}
+						this.timestampOfLastUsedJoyconInput = joycon.timestamp
+					}
+				} else if (joycon.mode === 'LR') {
+					// L + R mode
+					// get the first one that is moving outside of the deadband, prioritizing the L controller
```

Comment:
There's still the tiiiiiiniest small edgecas:

When, and only when, the R controller is stuck and gets disconnected. If giving the L some input, it takes over as expected, but when releasing the L again, the R will continue scrolling. This comes from both sides having their timestamp updated at the same time when paired, which they appear to be for a second or two before going to the single L config.

This can only happen when going from LR to L, with R stuck.

This would not have happened with a > comparison at :232, but tests shows that the UX of > is broken, as @jstarpl expected when going on a steady pace.

Reply:
I think this is the best we both could and should do.

Reply:
Well, I can't say that I experienced it: I just predicted that it would happen, based on the algorithm we had.